src/backend/distributed/master/master_create_shards.c

```diff
@@ -158,7 +159,14 @@ master_create_worker_shards(PG_FUNCTION_ARGS)
 	/* set shard storage type according to relation type */
 	if (relationKind == RELKIND_FOREIGN_TABLE)
 	{
-		shardStorageType = SHARD_STORAGE_FOREIGN;
+		if (cstoreTable)
```

Comment:
We can define cstoreTable here because we don't use it anywhere else. Similarly for other changes.